packages/repository/src/decorators/metadata.ts

```diff
@@ -0,0 +1,41 @@
+import {InspectionOptions, MetadataInspector} from '@loopback/context';
+import {MODEL_KEY, MODEL_PROPERTIES_KEY} from '../';
+import {ModelDefinition, PropertyDefinition} from '../../index';
+import * as _ from 'lodash';
+
+export class ModelMetadataHelper {
+  /**
+   * A utility function to simplify retrieving metadata from a target model and
+   * its properties.
+   * @param target The class from which to retrieve metadata.
+   * @param options An options object for the MetadataInspector to customize
+   * the output of the metadata retrieval functions.
+   */
+  static getModelMetadata(target: Function, options?: InspectionOptions) {
+    const classDef = MetadataInspector.getClassMetadata(
+      MODEL_KEY,
```

Comment:
I think we should use a different key to cache the fully built model definition.